src/test/rbf_tests.cpp

```diff
@@ -0,0 +1,225 @@
+    CTransactionRef tx4 = make_tx(/*output_values=*/ {999 * CENT}, /*inputs=*/ {tx3});
+    pool.addUnchecked(entry.Fee(high_fee).FromTx(tx4));
+
+    // Create a parent tx5 and child tx6 where both have very low fees
+    CTransactionRef tx5 = make_tx(/*output_values=*/ {1099 * CENT}, /*inputs=*/ {m_coinbase_txns[2]});
+    pool.addUnchecked(entry.Fee(low_fee).FromTx(tx5));
+    CTransactionRef tx6 = make_tx(/*output_values=*/ {1098 * CENT}, /*inputs=*/ {tx3});
+    pool.addUnchecked(entry.Fee(low_fee).FromTx(tx6));
+    // Make tx6's modified fee much higher than its base fee. This should cause it to pass
+    // the fee-related checks despite being low-feerate.
+    pool.PrioritiseTransaction(tx6->GetHash(), 1 * COIN);
+
+    // Two independent high-feerate transactions, tx7 and tx8
+    CTransactionRef tx7 = make_tx(/*output_values=*/ {999 * CENT}, /*inputs=*/ {m_coinbase_txns[3]});
+    pool.addUnchecked(entry.Fee(high_fee).FromTx(tx7));
+    CTransactionRef tx8 = make_tx(/*output_values=*/ {999 * CENT}, /*inputs=*/ {m_coinbase_txns[4]});
+    pool.addUnchecked(entry.Fee(high_fee).FromTx(tx8));
+
+    const auto entry1 = pool.GetIter(tx1->GetHash()).value();
+    const auto entry2 = pool.GetIter(tx2->GetHash()).value();
+    const auto entry3 = pool.GetIter(tx3->GetHash()).value();
+    const auto entry4 = pool.GetIter(tx4->GetHash()).value();
+    const auto entry5 = pool.GetIter(tx5->GetHash()).value();
+    const auto entry6 = pool.GetIter(tx6->GetHash()).value();
+    const auto entry7 = pool.GetIter(tx7->GetHash()).value();
+    const auto entry8 = pool.GetIter(tx8->GetHash()).value();
+
+    BOOST_CHECK_EQUAL(entry1->GetFee(), normal_fee);
+    BOOST_CHECK_EQUAL(entry2->GetFee(), normal_fee);
+    BOOST_CHECK_EQUAL(entry3->GetFee(), low_fee);
+    BOOST_CHECK_EQUAL(entry4->GetFee(), high_fee);
+    BOOST_CHECK_EQUAL(entry5->GetFee(), low_fee);
+    BOOST_CHECK_EQUAL(entry6->GetFee(), low_fee);
+    BOOST_CHECK_EQUAL(entry7->GetFee(), high_fee);
+    BOOST_CHECK_EQUAL(entry8->GetFee(), high_fee);
+
+    CTxMemPool::setEntries set_12_normal{entry1, entry2};
+    CTxMemPool::setEntries set_34_cpfp{entry3, entry4};
+    CTxMemPool::setEntries set_56_low{entry5, entry6};
+    CTxMemPool::setEntries all_entries{entry1, entry2, entry3, entry4, entry5, entry6, entry7, entry8};
+    CTxMemPool::setEntries empty_set;
+
+    const auto unused_txid{GetRandHash()};
+
+    // Tests for PaysMoreThanConflicts
+    // These tests use feerate, not absolute fee.
+    BOOST_CHECK(PaysMoreThanConflicts(/*iters_conflicting=*/set_12_normal,
+                                      /*replacement_feerate=*/CFeeRate(entry1->GetModifiedFee() + 1, entry1->GetTxSize() + 2),
+                                      /*txid=*/unused_txid).has_value());
+    // Replacement must be strictly greater than the originals.
+    BOOST_CHECK(PaysMoreThanConflicts(set_12_normal, CFeeRate(entry1->GetModifiedFee(), entry1->GetTxSize()), unused_txid).has_value());
```

Comment:
Changing the check in `PaysMoreThanConflict` to anything else than `<=` (ie `<`, `>`, `>=`, `==`) would fail the test already.

Changing to `=` would not compile.

Reply:
Yeah, I think the off-by-one error is accounted for on these lines. i.e. same fee fails, same fee + 1 succeeds.